src/sknnr/transformers/_cca.py

```diff
@@ -187,6 +191,14 @@ def coefficients(self):
         right = Q.T @ self._U
         return np.linalg.lstsq(R, right, rcond=None)[0]
 
+    def projector(self, n_components=None):
+        if n_components is None:
+            return self.coefficients @ self.axis_weights
```

Comment:
What do you think about replacing this early return with `n_components = self.max_components`? In other words:

```python
def projector(self, n_components=None):
        if n_components is None:
            n_components = self.max_components

        return (
            self.coefficients[:, :n_components]
            @ self.axis_weights[:n_components, :n_components]
        )
```

I find that a little more readable because the same return will be used in both cases, but maybe the fact that the indexing would have no effect outweighs that benefit?

Reply:
I agree, much cleaner!